checkout/views.py

```diff
@@ -0,0 +1,190 @@
+			order = order_form.save()
+			for item_id, item_data in cart.items():
+				try:
+					product = Product.objects.get(id=item_id)
+					# If we're working w/ an item with no sizes, item_data will be the quantity
+					if isinstance(item_data, int):
+						order_line_item = OrderLineItem(
+							order = order, 
+							product = product, 
+							quantity = item_data
+						)
+						order_line_item.save()
+					# Otherwise we need to add order lineitems for each size
+					else:
+						for size, quantity in item_data['items_by_size'].items():
+							order_line_item = OrderLineItem(
+								order = order, 
+								product = product, 
+								quantity = quantity,
+								product_size = size
+							)
+							order_line_item.save()
+				except Product.DoesNotExist:
+					messages.error(request, (
+						"One of the products in your bag wasn't found in our database. "
+						"Please call us for assistance! Your card was not charged.")
+					)
+					order.delete()
+					return redirect(reverse('view_cart'))
+
+			###################################
+			# TODO: Actual Stripe charge here #
+			###################################
+
+			# Save the info to the user's profile if all is well
+			request.session['save_info'] = 'save-info' in request.POST
+
+			return(redirect(reverse('checkout_success', args=[order.order_number])))
+		else:
+			messages.error(request, 'There was an error with your form. Please double check your information.')
+	else:
+		cart = request.session.get('cart', {})
+		if not cart:
+			messages.error(request, f"You don't have anything in your bag at the moment. Try adding some products before checking out.")
+			return redirect('products')
+		# Attempt to prefill the form with any info the user maintains in their profile
+		try:
+			profile = UserProfile.objects.get(user=request.user)
+			order_form = OrderForm(initial={
+				'full_name': profile.user.get_full_name(),
+				'email': profile.user.email,
+				'phone_number': profile.default_phone_number,
+				'country': profile.default_country,
+				'postcode': profile.default_postcode,
+				'town_or_city': profile.default_town_or_city,
+				'street_address1': profile.default_street_address1,
+				'street_address2': profile.default_street_address2,
+				'county': profile.default_county,
+			})
+		except UserProfile.DoesNotExist:
+			order_form = OrderForm()
+
+	stripe_public_key = settings.STRIPE_PUBLIC_KEY
+
+	if not stripe_public_key:
+		messages.warning(request, 'Stripe public key is missing. Did you forget to set it in your environment?')
+		
+	template = 'checkout/checkout.html'
+	context = {
+		'stripe_public_key': stripe_public_key,
+		'order_form': order_form,
+	}
+
+	return render(request, template, context)
+
+
+def checkout_success(request, order_number):
+	# Check the session to see if the user wanted to save their info
+	save_info = request.session.get('save_info')
+
+	order = Order.objects.get(order_number=order_number)
+	no_errors = True
```

Comment:
This is wishful thinking ;)

Reply:
I thought this is how you debug? Just set a variable to get rid of all the errors :)